lib/Frontend/CompilerInvocation.cpp

```diff
@@ -47,7 +47,25 @@ static void updateRuntimeLibraryPath(SearchPathOptions &SearchPathOpts,
   llvm::sys::path::append(LibPath, getPlatformNameForTriple(Triple));
   SearchPathOpts.RuntimeLibraryPath = LibPath.str();
 
-  llvm::sys::path::append(LibPath, Triple.getArchName());
+  // The linux provided triple for ARM contains a trailing 'l'
+  // denoting little-endian.  This is not used in the path for
+  // libraries.  LLVM matches these SubArchTypes to the generic
+  // ARMSubArch_v7 (for example) type.  If that is the case,
+  // use the base of the architecture type in the library path.
+  if (Triple.isOSLinux()) {
```

Comment:
…although another possible answer would be to actually *use* `armv7l` as the name of the standard library, in case you ever wanted it to coexist with an `armv7b`.